File: app/models/network_cache/validator.rb
module NetworkCache
  class Validator
    include FilePath

    def self.from(testing_ground, opts = {})
      new(testing_ground, opts)
    end

    def initialize(testing_ground, opts = {})
      @testing_ground = testing_ground
      @opts = opts
    end

    def valid?(force_clear = false)
      return false if force_clear

      @testing_ground.topology.each_node.all? do |node|
        File.exists?(file_name(node[:name]))
      end
    end
  end
end

File: app/assets/javascripts/topology/validator.js
/*globals ETHelper,Topology,Validator*/

Topology.Validator = (function () {
    'use strict';

    return {
        isValid: function (graph) {
            var valid = true,
                lastInvalidNode;

            ETHelper.eachNode([graph.data], function (node) {
                if (!node.name || node.name === '') {
                    node.errors = { name: "name can't be blank" };

                    lastInvalidNode = node;

                    valid = false;
                } else {
                    delete node.errors;
                }
            });

            graph.update(lastInvalidNode || graph.data);
            window.TopologyEditor.graphData.setData(graph.data);

            if (lastInvalidNode) {
                window.TopologyEditor.graphEditor.focusId = lastInvalidNode.id;
                window.TopologyEditor.form.show(lastInvalidNode);
                window.TopologyEditor.nodeInterface.reposition();
            }


            return valid;
        }
    };
}());
